cdap-ui/app/directives/log-viewer/log-viewer.html

```diff
@@ -14,38 +14,135 @@
+          <div class="stat-count total-num">
+            <span class="stat-count-number">{{LogViewer.totalCount}}</span></div>
+      </div>
+      <div class="log-stats-errors log-stats text-center">
+        <span class="stat-name log-stats-errors-text">Error</span>
+          <div class="stat-count errors-num"><span class='stat-count-number'>{{LogViewer.errorCount}}</span></div>
       </div>
-      <div class="text-center" ng-if="loadingNext">
-        <span class="fa fa-refresh fa-spin"></span> Loading Logs
+      <div class="log-stats-warnings log-stats text-center">
+        <span class="stat-name log-stats-warnings-text">Warning</span>
+        <div class="stat-count warnings-num">
+          <span class='stat-count-number'>{{LogViewer.warningCount}}</span>
+        </div>
+      </div>
+    </div>
+
+    <div class="log-actions">
+      <div class="log-action-container">
+        <button class="log-action-btn">View Raw</button>
+        <button class="log-action-btn">Download All</button>
+        <div class="search">
+          <input type='text' class="log-searchbar" placeholder="Search" ng-model="searchText">
+          <span class="fa fa-search"></span>
+        </div>
+        <i class="fa fa-expand" aria-hidden="true" ng-click=""></i>
       </div>
     </div>
 
   </div>
+
+  <!-- LOGS TIMELINE -->
+  <div class="logs-timeline">
+    <div class="timeline-buffer">
+    </div>
+  </div>
+
+  <!-- LOGS TABLE -->
+  <div class="logs-table">
+    <table class="table table-bordered">
+      <thead>
+        <th class="time"
+            ng-if="!LogViewer.isMessageExpanded && LogViewer.configOptions.time">
+          Time
+        </th>
+
+        <th class="level"
+            ng-if="!LogViewer.isMessageExpanded && LogViewer.configOptions.level">
+          <span uib-dropdown>
+            <span uib-dropdown-toggle>Level <span class="fa fa-caret-down"></span></span>
+            <ul uib-dropdown-menu ng-click="$event.stopPropagation();">
+              <li ng-repeat="option in LogViewer.logEvents">
+                <input type="checkbox" ng-click="LogViewer.includeEvent(option)"><span ui-sref='test'>{{option}}</span></li>
+              </li>
+            </ul>
+          </span>
+        </th>
+
+        <th class="source"
+            ng-if="!LogViewer.isMessageExpanded && LogViewer.configOptions.source">
+          Source
+        </th>
+
+        <th class="message"
+            ng-if="LogViewer.configOptions.message"
+            ng-class="{'expanded': LogViewer.isMessageExpanded }">
+          <span class="fa"
+                ng-class="{ 'fa-arrow-left': !LogViewer.isMessageExpanded,
+                  'fa-arrow-right': LogViewer.isMessageExpanded}"
+                ng-click="LogViewer.isMessageExpanded = !LogViewer.isMessageExpanded"></span>
+          <span>Message</span>
+        </th>
+      </thead>
+
+      <tbody>
+        <tr ng-repeat="log in LogViewer.data | filter:LogViewer.eventFilter | filter:searchText">
```

Comment:
shouldn't use angular filter for this, it is going to be way too slow and costly. Angular filter is evaluated every digest cycle. We could have thousands of lines for the logs, so it's not ideal